packages/grafana-data/src/field/displayProcessor.ts

```diff
@@ -44,7 +45,8 @@ export function getDisplayProcessor(options?: DisplayProcessorOptions): DisplayP
 
   const field = options.field as Field;
   const config = field.config ?? {};
-  const { palette } = options.theme.visualization;
+  const theme = options.theme || createTheme({ colors: { mode: 'dark' } });
```

Comment:
Consistent with comment on line 25

Reply:
It used to be optional, then it was marked as required in https://github.com/grafana/grafana/pull/33502. There's no description in that PR and I have no context, but I wonder if that was intentional to make it a required param and the comment is wrong 🤔  ?

Reply:
I guess it's because the display processor in it's shape actually has to have a theme provided to get the correct mappings calculations for instance. But with this default fallback it may make sense.

Reply:
It would be better if we keep it as required so it needs to be passed. We are currently looking into ways to move the "lazy creation" of the theme  context from `@grafana/data` to core when the app starts.